### PR TITLE
Upgrade to Angular 5

### PR DESCRIPTION
**Pull request by sendilkumarn**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,28 +20,29 @@
   "typings": "index.d.ts",
   "homepage": "https://jhipster.github.io",
   "peerDependencies": {
-    "@angular/core": "^4.1.3",
-    "@angular/http": "^4.1.3"
+    "@angular/core": "5.0.0",
```
Comment (deepu105) [inline package.json]: use `^` versions as these are peerDependencies else we would have to keep them in sync. These dependencies will be decided by the generated app

File: package.json [hunk 2]
```diff
@@ -20,28 +20,29 @@
...
-    "@angular/common": "^4.1.3",
-    "@angular/compiler": "^4.1.3",
-    "@angular/compiler-cli": "^4.1.3",
-    "@angular/core": "^4.1.3",
-    "@angular/forms": "^4.1.3",
-    "@angular/http": "^4.1.3",
-    "@angular/platform-browser": "^4.1.3",
-    "@angular/platform-browser-dynamic": "^4.1.3",
-    "@angular/platform-server": "^4.1.3",
+    "@angular/animations": "5.0.0",
```
Comment (deepu105) [inline package.json]: use `^` versions here as well so that libs are in sync

File: tslint.json [hunk 3]
```diff
@@ -106,9 +106,6 @@
         "use-life-cycle-interface": true,
         "use-pipe-transform-interface": false,
         "component-class-suffix": true,
-        "directive-class-suffix": true,
-        "no-access-missing-member": true,
-        "templates-use-public": true,
-        "invoke-injectable": true
```
Comment (deepu105) [inline tslint.json]: are the rules removed or are they giving lint issues?

Comment (sendilkumarn) [reply tslint.json]: yeah

**Pull request merged**